src/fluree/db/json_ld/credential.cljc

```diff
@@ -58,26 +58,27 @@
 
 (defn generate
   "Generate a VerifiableCredential given a subject and some issuer opts."
-  ([credential-subject private] (generate credential-subject private
-                                          (did/private->did-map private)))
+  ([credential-subject private]
+   (generate credential-subject private (did/private->did-map private)))
   ([credential-subject private did]
    (go-try
-     (let [proof #?(:clj (create-proof (-> credential-subject
-                                           jld-processor/canonize
-                                           crypto/sha2-256)
-                                       (did/encode-did-key (:public did))
-                                       private)
-                    :cljs (let [canonicalized (<p! (jld-processor/canonize credential-subject))]
-                            (create-proof (crypto/sha2-256 canonicalized)
-                                          (did/encode-did-key (:public did))
-                                          private)))]
+     (let [canonicalized #?(:clj (jld-processor/canonize credential-subject)
+                            :cljs (<p! (jld-processor/canonize credential-subject)))
+
+           ;; TODO: assert this once our credential subjects are proper json-ld
+           ;; _ (when (= "" canonicalized) (throw (ex-info "Unsupported credential subject" {:credential-subject credential-subject})))
+
+           did-key (did/encode-did-key (:public did))
+           proof (create-proof (crypto/sha2-256 canonicalized)
+                               did-key
+                               private)]
        {"@context"          "https://www.w3.org/2018/credentials/v1"
         "id"                ""
         "type"              ["VerifiableCredential" "CommitProof"]
         "issuer"            (:id did)
         "issuanceDate"      (util/current-time-iso)
         "credentialSubject" credential-subject
-        "proof" proof}))))
+        "proof"             proof}))))
 
 (defn verify
   "Takes a credential and returns the credential subject and issuer id if it verifies. If
```

Comment:
I know this is unrelated to this pull request, but it looks like this docstring is out of date. Could you update it to say that this function returns `nil` if there's no signature?